test/built-ins/Promise/resolve-poisoned-then-deferred.js

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+description: >
+    Resolving with an object with a "poisoned" `then` property after invocation
+    of the executor function
+es6id: 25.4.3.1
+info: >
+    [...]
+    8. Let resolvingFunctions be CreateResolvingFunctions(promise).
+    9. Let completion be Call(executor, undefined,
+       «resolvingFunctions.[[Resolve]], resolvingFunctions.[[Reject]]»).
+
+    25.4.1.3.2 Promise Resolve Functions
+    7. If Type(resolution) is not Object, then
+       a. Return FulfillPromise(promise, resolution).
+---*/
+
+var value = {};
+var poisonedThen = Object.defineProperty({}, 'then', {
+  get: function() {
+    throw value;
+  }
+});
+var promise = new Promise(function(_resolve) {
+  resolve = _resolve;
```

Comment:
In strict mode, this doesn't create a global variable and the call at the end of the file fails because of that.